torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -341,6 +341,10 @@ class TORCH_API ProcessGroupNCCL : public Backend {
     // Clone of opTimeout_ from ProcessGroupNCCL.
     std::chrono::milliseconds opTimeout_;
 
+    // timeout to be reduced after this work finishes.
```

Comment:
more verbose explanation

"ephemeral timeouts are owned by exactly one work, and decay after that work completes.  There may be more than one ephemeral timeout active at the same time." 

or something like hat